File: interpreter/src/test/java/cambio/tltea/interpreter/BehaviorInterpreterTest.java
package cambio.tltea.interpreter;

import cambio.tltea.interpreter.nodes.cause.EventActivationListener;
import cambio.tltea.parser.core.ASTNode;
import cambio.tltea.parser.core.temporal.TimeInstance;
import cambio.tltea.parser.mtl.generated.MTLParser;
import cambio.tltea.parser.mtl.generated.ParseException;
import org.junit.jupiter.api.Test;

class BehaviorInterpreterTest {
    @Test
    void debugTest() throws ParseException {
        String input = "((P)&(C)&(D))->(Q)";
        ASTNode ast = new MTLParser(input).parse();
        BehaviorInterpreter behaviorInterpreter = new BehaviorInterpreter();
        var result = behaviorInterpreter.interpret(ast);

        result.getTriggerNotifier().subscribe((eventName, args) -> {
            System.out.println(eventName);
        });

        result.getListeners().forEach(eventActivationListener -> eventActivationListener.activate(new TimeInstance(0)));

        System.out.println(result);
    }
}

File: interpreter/src/test/java/cambio/tltea/interpreter/BehaviorInterpreterTest.kt
package cambio.tltea.interpreter

import cambio.tltea.interpreter.Interpreter.interpretAsBehavior
import cambio.tltea.parser.core.temporal.TimeInstance
import cambio.tltea.parser.mtl.generated.MTLParser
import cambio.tltea.parser.mtl.generated.ParseException
import org.junit.jupiter.api.Test

internal class BehaviorInterpreterTest {
    @Test
    @Throws(ParseException::class)
    fun debugTest() {
        val input = "((P)&(!!(C))|!(D))->((Q)==1234)"
        val ast = MTLParser(input).parse()
        val result = interpretAsBehavior(ast)
        result.activateProcessing();


        result.triggerManager.subscribeEventListener {
            println("hi $it")
        }

        result.triggerManager.eventActivationListeners
            .filter { !it.eventName.contains("F") }
            .forEach { it.activate(TimeInstance(0.0)) }
        result.triggerManager.eventActivationListeners
            .filter { it.eventName.contains("F") }
            .forEach { it.activate(TimeInstance(0.0)) }

        println(result)
    }
}